hera_pspec/uvwindow.py

```diff
@@ -0,0 +1,1320 @@
+            1D float array of ascending k_perp bin centers in [h] Mpc^-1 units.
+            Used for cylindrical binning,
+            Make sure the values are consistent with :attr:`little_h`.
+        kpara_bins : array-like astropy.quantity with units, optional.
+            1D float array of ascending k_parallel bin centers.
+            Used for cylindrical binning.
+            Make sure the values are consistent with :attr:`little_h`.
+        overwrite : bool, optional
+            Whether to overwrite output file if it exists. Default: False.
+        """
+        # Check output
+        if os.path.exists(filepath) and overwrite is False:
+            raise IOError("{} exists, not overwriting...".format(filepath))
+        elif os.path.exists(filepath) and overwrite is True:
+            print("{} exists, overwriting...".format(filepath))
+            os.remove(filepath)
+
+        nbls = len(bl_lens)  # number of redudant groups
+        bl_lens = np.array(bl_lens)
+        if bl_weights is not None:
+            # check consistency of baseline-related inputs
+            assert len(bl_weights) == nbls, "bl_weights and bl_lens "\
+                                            "must have same length"
+            bl_weights = np.array(bl_weights)
+        else:
+            # each baseline length has weight one
+            bl_weights = np.ones(nbls)
+
+        # k-bins for cylindrical binning
+        if kperp_bins is None or len(kperp_bins) == 0:
+            # define default kperp bins, making sure all values probed by
+            # bl_lens are included and there is no over-sampling
+            kperp_bins = self.get_kperp_bins(bl_lens)
+        else:
+            self.check_kunits(kperp_bins)
+        kperp_bins = np.array(kperp_bins.value)
+        nbins_kperp = kperp_bins.size
+
+        if kpara_bins is None or len(kpara_bins) == 0:
+            # define default kperp bins, making sure all values probed by freq
+            # array are included and there is no over-sampling
+            kpara_bins = self.get_kpara_bins(self.freq_array,
+                                             self.little_h, self.cosmo)
+        else:
+            self.check_kunits(kpara_bins)
+        kpara_bins = np.array(kpara_bins.value)
+        nbins_kpara = kpara_bins.size
+
+        # COMPUTE THE WINDOW FUNCTIONS
+
+        # get cylindrical window functions for each baseline length considered
+        # as a function of (kperp, kpara)
+        # the kperp and kpara bins are given as global parameters
+        cyl_wf = np.zeros((nbls, self.Nfreqs, nbins_kperp, nbins_kpara))
+        for ib in range(nbls):
+            cyl_wf[ib, :, :, :] = self.get_cylindrical_wf(bl_lens[ib],
+                                                          kperp_bins*self.kunits,
+                                                          kpara_bins*self.kunits)*bl_weights[ib]
+
+        # Write file
+        with h5py.File(filepath, 'w') as f:
```

Comment:
things like ints and float are typically better written into the `.attrs` group a dataset.